src/operator/nn/mkldnn/mkldnn_batch_norm-inl.h

```diff
@@ -241,25 +176,30 @@ void MKLDNNBatchNormForward(const OpContext &ctx, const BatchNormParam &param,
     DType* weight_buf = reinterpret_cast<DType *>(weight_mem.get_data_handle());
 
     nnvm::dim_t channels_ = data.shape()[1];
-    CHECK(weight_mem.get_primitive_desc().get_size() == channels_ * sizeof(DType) * 2);
+    CHECK(weight_mem.get_desc().get_size() == channels_ * sizeof(DType) * 2);
     DType* weight_ptr = gamma.data().dptr<DType>();
     DType* bias_ptr = beta.data().dptr<DType>();
     if (!param.fix_gamma) {
       memcpy(weight_buf, weight_ptr, sizeof(weight_buf[0]) * channels_);
       memcpy(&weight_buf[channels_], bias_ptr, sizeof(weight_buf[0]) * channels_);
     } else if (IsBNWriting(req[batchnorm::kGamma])) {
       for (int i = 0; i < channels_; i++) {
-        weight_buf[i] = (DType)1.0f;
-        weight_ptr[i] = (DType)1.0f;
+        weight_buf[i] = static_cast<DType>(1.0f);
+        weight_ptr[i] = static_cast<DType>(1.0f);
         weight_buf[channels_ + i] = bias_ptr[i];  // bias
       }
     } else {
       for (int i = 0; i < channels_; i++) {
-        weight_buf[i] = (DType)1.0f;
+        weight_buf[i] = static_cast<DType>(1.0f);
         weight_buf[channels_ + i] = bias_ptr[i];  // bias
       }
     }
 
+    std::unordered_map<int, mkldnn::memory> net_args;
```

Comment:
Use mkldnn_args_map_t instead of std::unordered_map<int, mkldnn::memory>

Reply:
OK